test/test/CompilerTest.scala

```diff
@@ -465,7 +492,7 @@ abstract class CompilerTest {
 
   /** Reads the existing files for the given test source if any. */
   private def getExisting(dest: Path): ExistingFiles = {
-    val content: Option[Option[String]] = processFileDir(dest, f => f.safeSlurp, d => Some(""))
+    val content: Option[Option[String]] = processFileDir(dest, f => try Some(f.slurp("UTF8")) catch {case io: java.io.IOException => Some(io.toString())}, d => Some(""))
```

Comment:
This ugly thing definitely need be factored out (also used a couple lines below).

What does this do, anyway? Shouldn't `safeSlurp` be responsible of this safety net?